firebase-firestore/src/main/java/com/google/firebase/firestore/util/Executors.java

```diff
@@ -29,6 +40,39 @@
   /** An executor that executes the provided runnable immediately on the current thread. */
   public static final Executor DIRECT_EXECUTOR = Runnable::run;
 
+  /**
+   * An executor that runs tasks in parallel on Android's AsyncTask.THREAD_POOL_EXECUTOR.
+   *
+   * <p>Unlike the main THREAD_POOL_EXECUTOR, this executor manages its own queue of tasks and can
+   * handle an unbounded number of pending tasks.
+   */
+  public static final Executor BACKGROUND_EXECUTOR =
+      new Executor() {
```

Comment:
Please don't make major components we're going to reason about like this anonymous classes. Make it a `private static final class` with a name. This will make stack traces much, much saner.